app/scripts/views/confirm_sign_in.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var BaseView = require('views/base');
+  var Cocktail = require('cocktail');
+  var Constants = require('lib/constants');
+  var ExperimentMixin = require('views/mixins/experiment-mixin');
+  var FormView = require('views/form');
+  var ResendMixin = require('views/mixins/resend-mixin');
+  var ResumeTokenMixin = require('views/mixins/resume-token-mixin');
+  var ServiceMixin = require('views/mixins/service-mixin');
+  var Template = require('stache!templates/confirm_sign_in');
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'confirm_sign_in',
+
+    // used by unit tests
+    VERIFICATION_POLL_IN_MS: Constants.VERIFICATION_POLL_IN_MS,
+
+    initialize: function () {
+      // Account data is passed in from sign up and sign in flows.
+      // It's important for Sync flows where account data holds
+      // ephemeral properties like unwrapBKey and keyFetchToken
+      // that need to be sent to the browser.
+      this._account = this.user.initAccount(this.model.get('account'));
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    context: function () {
+      var email = this.getAccount().get('email');
+
+      return {
+        email: email,
+        isOpenGmailButtonVisible: this._isOpenGmailButtonVisible(),
+        safeEmail: encodeURIComponent(email)
+      };
+    },
+
+    _isOpenGmailButtonVisible: function () {
```

Comment:
Is the gmail copy/paste code from elsewhere? If so, we should extract it into a mixin and share it among the modules. That'd be a good refactoring to do in a separate PR that we can merge before everything else.